compiler/llvm/clangUtil.cpp

```diff
@@ -2235,6 +2335,78 @@ void configurePMBuilder(PassManagerBuilder &PMBuilder, bool forFunctionPasses, i
   // TODO: we might need to call TargetMachine's addEarlyAsPossiblePasses
 }
 
+#ifndef LLVM_USE_OLD_PASSES
+
+static void registerDumpIrExtensions(PassBuilder& PB);
+
+static void runModuleOptPipeline(bool addWideOpts) {
+  GenInfo* info = gGenInfo;
+
+  LoopAnalysisManager LAM;
+  FunctionAnalysisManager FAM;
+  CGSCCAnalysisManager CGAM;
+  ModuleAnalysisManager MAM;
+
+  PassInstrumentationCallbacks PIC;
+  StandardInstrumentations SI(
+#if HAVE_LLVM >= 160
+                              info->llvmContext,
+#endif
+                              /* DebugLogging */ false);
+  SI.registerCallbacks(PIC, &FAM);
+
+  chpl::optional<PGOOptions> PGOOpt;
+  PassBuilder PB(info->targetMachine, createPipelineOptions(false),
+                 PGOOpt, &PIC);
+
+
+  // some FAM add-ins
+  std::unique_ptr<TargetLibraryInfoImpl> TLII(
+      new TargetLibraryInfoImpl(llvm::Triple(info->targetMachine->getTargetTriple())));
+  // TODO: handle CodeGenOptions::LIBMVEC etc to support
+  // vectorizing the math library.
```

Comment:
I'm not seeing it. If you are talking about `llvm::FastMathFlags`, we should be setting that per-function with the IRBuilder (with either pass manager).

Reply:
I do mean `llvm::FastMathFlags`, the code I was looking at is this: https://github.com/chapel-lang/chapel/blob/ebfe637f856fadf96d1b133ecd6cb4b7f9c13fc2/compiler/llvm/clangUtil.cpp#L2264-L2282

This code looks like it's setting up these flags for the IRBuilder to use later; I don't understand the context of it very well but called it out since I couldn't find an equivalent in the new code.

Reply:
Right `prepareCodegenLLVM` is called in `codegenPartTwo` which runs whether or not we are using the new pass manager.